### PR TITLE
[Train/Tune] De-flake CI tests

### LINKED ISSUE #41150: [CI] `linux://doc/source/tune/examples:tune-pytorch-cifar` is failing/flaky on master.
- 329a65364fbb95b7d30b0c476fde463b7b08603b FAILED [Buildkite :train: ml: doc tests](https://buildkite.com/ray-project/postmerge/builds/1677#018bd096-fa68-433a-a899-f4d83ee533c9)
- c39bd4f6dcbd26bc3a615ed2dce3c68ebca1f59f FAILED [Buildkite :train: ml: doc tests](https://buildkite.com/ray-project/postmerge/builds/1676#018bd089-4230-4896-a5f8-3286fabe7c27)
- e667cd7a29fed968c9fc5a73b66a217cdfe62e24 FAILED [Buildkite :train: ml: doc tests](https://buildkite.com/ray-project/postmerge/builds/1675#018bd0a9-2592-41d0-aa29-791aac59ba13)
- b96a79edcaa269837cf60ede8dbe4ac8edc8c757 FAILED [Buildkite :train: ml: doc tests](https://buildkite.com/ray-project/postmerge/builds/1673#018bd087-465d-47bb-9253-232fb7c366e8)

....
Generated from flaky test tracker. Please do not edit the signature in this section.
DataCaseName-linux://doc/source/tune/examples:tune-pytorch-cifar-END
....

### LINKED ISSUE #41179: [CI] `linux://python/ray/tune:test_commands` is failing/flaky on master.
- 227fe5dba383d2374b91309dd644ae9c1b886b00 FAILED [Buildkite :train: ml: flaky tests](https://buildkite.com/ray-project/postmerge/builds/1686#018bd442-f583-4182-9731-c22615a72cad)
- cc93df5824ac38f44e9f64723bc435ecbc5fbd16 FAILED [Buildkite :train: ml: flaky tests](https://buildkite.com/ray-project/postmerge/builds/1600#018bab8f-45c0-4c59-81dd-3a77b7bd717b)

....
Generated from flaky test tracker. Please do not edit the signature in this section.
DataCaseName-linux://python/ray/tune:test_commands-END
....

### LINKED ISSUE #41180: [CI] `linux://python/ray/train:torch_exp_tracking_mlflow` is failing/flaky on master.
- 3c35a73ae6227cc5504af34276ccaedad3812059 FAILED [Buildkite :train: ml: flaky tests](https://buildkite.com/ray-project/postmerge/builds/1681#018bd17b-0520-4aab-b445-7b9ddf07b0bc)
- ce550561c5580edc32d30a296e997e3a8a208ad9 FAILED [Buildkite :train: ml: flaky tests](https://buildkite.com/ray-project/postmerge/builds/1668#018bcf80-2cdf-4cec-a5be-1acf94538d68)
- 00a1dcbcb818afb9fe2e49ae9f76ab07f19e5426 FAILED [Buildkite :train: ml: flaky tests](https://buildkite.com/ray-project/postmerge/builds/1632#018bb66e-c600-48bd-b0de-5a76293db524)
- c49c3890d727da6308deaee527d2611de7205a63 FAILED [Buildkite :train: ml: flaky tests](https://buildkite.com/ray-project/postmerge/builds/1606#018bacd1-ec62-4ee4-a2a5-616253522433)
- 451f60b7ff18376719910008f3ee71c0d383d2bd FAILED [Buildkite :train: ml: flaky tests](https://buildkite.com/ray-project/postmerge/builds/1604#018bac32-f6ec-4334-a19e-4e61301f6db8)
- 130a1888887f84c96d16c4d71268a4784a6a02df FAILED [Buildkite :train: ml: flaky tests](https://buildkite.com/ray-project/postmerge/builds/1593#018baae8-72db-43b9-886e-f883d88c8ea5)
- 67ec953c17e676cfd7956ce6ca8e0a024dd9c9c7 FAILED [Buildkite :train: ml: flaky tests](https://buildkite.com/ray-project/postmerge/builds/1580#018ba6e6-ae82-494e-9960-c78f0091631c)
- c3285a8f9ae4d04c25bf16f6d87e4a9ac13e420b FAILED [Buildkite :train: ml: flaky tests](https://buildkite.com/ray-project/postmerge/builds/1568#018ba44c-7c1c-47ea-97de-4d4af0a40b7b)

....
Generated from flaky test tracker. Please do not edit the signature in this section.
DataCaseName-linux://python/ray/train:torch_exp_tracking_mlflow-END
....

### LINKED ISSUE #41181: [CI] `linux://python/ray/tests/xgboost:simple_tune` is failing/flaky on master.
- 89bc3dc6fac99be7f3d0cefa54608e70ed1646ed FAILED [Buildkite :train: ml: library integration tests](https://buildkite.com/ray-project/postmerge/builds/1601#018babb7-61eb-4a20-b1d9-53b597ea6954)

....
Generated from flaky test tracker. Please do not edit the signature in this section.
DataCaseName-linux://python/ray/tests/xgboost:simple_tune-END
....

### LINKED ISSUE #41182: [CI] `linux://python/ray/train:torch_exp_tracking_wandb` is failing/flaky on master.
- 3c35a73ae6227cc5504af34276ccaedad3812059 FAILED [Buildkite :train: ml: flaky authentication tests](https://buildkite.com/ray-project/postmerge/builds/1681#018bd17b-0524-4934-84fe-2543f6daf795)
- 2c54abdb8aa0b3409667c7f4b1d06672d4b16b93 FAILED [Buildkite :train: ml: train authentication tests](https://buildkite.com/ray-project/postmerge/builds/1583#018ba6f5-b189-4918-b2cc-2936ac13e4d6)

....
Generated from flaky test tracker. Please do not edit the signature in this section.
DataCaseName-linux://python/ray/train:torch_exp_tracking_wandb-END
....

### LINKED ISSUE #41183: [CI] `linux://doc/source/train/examples/pytorch:convert_existing_pytorch_code_to_ray_train` is failing/flaky on master.
- 822f23716a32a7c74c0878d14d8ea7ee05e768de FAILED [Buildkite :train: ml: flaky tests](https://buildkite.com/ray-project/postmerge/builds/1689#018bd50f-4d8e-42d8-a193-41ff2bef1f91)
- 3c35a73ae6227cc5504af34276ccaedad3812059 FAILED [Buildkite :train: ml: flaky tests](https://buildkite.com/ray-project/postmerge/builds/1681#018bd17b-0520-4aab-b445-7b9ddf07b0bc)
- 62fa64079c6f9ecbadd2eb974d87e90b79612c32 FAILED [Buildkite :train: ml: flaky tests](https://buildkite.com/ray-project/postmerge/builds/1674#018bd05b-395f-478e-9121-031b303108af)
- 7a192627e1b6c0752d996d2ad7b34d115e539050 FAILED [Buildkite :train: ml: flaky tests](https://buildkite.com/ray-project/postmerge/builds/1662#018bcd7f-1671-4222-9c24-5694d0817066)
- 2d3865f2bf3f69e3596ec4478760d789be69c6f1 FAILED [Buildkite :train: ml: flaky tests](https://buildkite.com/ray-project/postmerge/builds/1637#018bb8e5-c02d-457f-a369-63ad413a7804)
- d5d0844a9d54353d62ce932fdd4230a1d0547502 FAILED [Buildkite :train: ml: flaky tests](https://buildkite.com/ray-project/postmerge/builds/1608#018bad5b-e84e-43ae-a688-bdf89248087a)
- 89bc3dc6fac99be7f3d0cefa54608e70ed1646ed FAILED [Buildkite :train: ml: flaky tests](https://buildkite.com/ray-project/postmerge/builds/1601#018babb7-61fa-437a-bc4e-dc6d8219c010)
- 2c54abdb8aa0b3409667c7f4b1d06672d4b16b93 FAILED [Buildkite :book: :airplane: Ray AIR examples](https://buildkite.com/ray-project/oss-ci-build-branch/builds/6806#018ba70f-267d-4be2-b516-146c40b85b8b)
- b07cd41175fff16ddb583afc67d3c78c7e28e72e FAILED [Buildkite :book: :airplane: Ray AIR examples](https://buildkite.com/ray-project/oss-ci-build-branch/builds/6801#018ba68f-bf85-454e-a517-cb573ed245c5)

....
Generated from flaky test tracker. Please do not edit the signature in this section.
DataCaseName-linux://doc/source/train/examples/pytorch:convert_existing_pytorch_code_to_ray_train-END
....

### PR DESCRIPTION
**Pull request by justinvyu**
## Why are these changes needed?

This PR fixes some flaky CI tests and removes them from the flaky list. The main problem is corrupted datasets due to multiple workers downloading to the same directory simultaneously. Other issues are flaky timeouts caused by external dataset download dependencies. This PR introduces mock data "smoke test" versions of the examples that don't depend external downloads.

## Related issue number

Closes https://github.com/ray-project/ray/issues/41183
Closes https://github.com/ray-project/ray/issues/41182
Closes https://github.com/ray-project/ray/issues/41181
Closes https://github.com/ray-project/ray/issues/41180
Closes https://github.com/ray-project/ray/issues/41179
Closes https://github.com/ray-project/ray/issues/41150

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Review by aslonnie — APPROVED** [review]
approval for test build file changes.

**Comment by can-anyscale** [comment]
premerge is failing?

File: ci/docker/ml.build.Dockerfile [hunk 1]
```diff
@@ -34,7 +34,7 @@ fi
 if [[ "$RAYCI_LIGHTNING_2" == "true" ]]; then
   pip uninstall -y pytorch-lightning
   # todo move to requirements-test.txt
-  pip install lightning==2.0.4 pytorch-lightning==2.0.4
```
Comment (matthewdeng) [inline ci/docker/ml.build.Dockerfile:37]: We can revert this and leave the test as flaky to unblock this PR and handle it separately.

File: python/ray/train/examples/experiment_tracking/torch_exp_tracking_wandb.py [hunk 2]
```diff
@@ -1,12 +1,14 @@
...
 
-# __start__
-# Run the following script with the WANDB_API_KEY env var set.
 import os
 
 import torch
 import wandb
+
+# __start__
+# Run the following script with the WANDB_API_KEY env var set.
```
Comment (matthewdeng) [inline python/ray/train/examples/experiment_tracking/torch_exp_tracking_wandb.py:10]: I think we need to disable isort for this file to keep this at the top.

**Review by matthewdeng — APPROVED** [review]

**Pull request merged**